src/hangar/arrayset.py

```diff
@@ -1093,7 +1084,8 @@ def items(self) -> Iterable[Tuple[str, Union[ArraysetDataReader, ArraysetDataWri
         Iterable[Tuple[str, Union[:class:`.ArraysetDataReader`, :class:`.ArraysetDataWriter`]]]
             returns two tuple of all all arrayset names/object pairs in the checkout.
         """
-        for asetN, asetObj in self._arraysets.items():
+        for asetN in list(self._arraysets.keys()):
```

Comment:
just curious: why an explicit `list()` here?

Reply:
Yeah.. I hate it too... It's because of the following situation 

```python
>>> for name, aset in co.arraysets.items():
>>>     del co.arraysets[name]

traceback RuntimeError
------------------------
RuntimeError: dictionary changed size during iteration
```
While I normally subscribe to the belief that "if you mutate a data structure while iterating over it, you are performing a mortal sin and deserve whatever happens to you", it's hangars responsibility to manager  usage of  this thing which kindof behaves like a `dict` and class simultaneously. Seemed unfair to put the implications of dealing with an implementation detail on the user when it can be fixed so trivially...